spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -135,4 +136,8 @@ public MessageConverter getMessageConverter() {
 	public void setMessageConverter(MessageConverter messageConverter) {
 		this.messageConverter = messageConverter;
 	}
+
+	public void setHeaderPatternsToMap(String... patterns) {
```

Comment:
Why do we lose a flexibility to let end-user to inject their own `HeaderMapper` implementation?
This pattern style is just a `PubSubHeaderMapper` responsiblity, therefore here we need just expect `HeaderMapper<Map<String, String>>`.